basyx/aas/adapter/rdf/rdf_serialization.py

```diff
@@ -0,0 +1,834 @@
+        """
+        reference = BNode()
+        self.graph.add((parent, objectProperty, reference))
+        self.graph.add((reference, RDF.type, self.aas["Reference"]))
+        self.graph.add((reference, self.aas["Reference/type"], self.aas[f"ReferenceTypes/{_generic.REFERENCE_TYPES[obj.__class__]}"]))
+        if obj.referred_semantic_id is not None:
+            self._reference_to_rdf(obj.referred_semantic_id, reference, self.aas["Reference/referredSemanticId"])
+        for aas_key in obj.key:
+            self._key_to_rdf(aas_key, reference)
+
+    def _qualifier_to_rdf(self, obj: model.Qualifier, parent: Union[URIRef, BNode]) -> None:
+        """
+        Serialization of objects of class :class:`~basyx.aas.model.base.Qualifier`
+        to a joint rdflib Graph object
+
+        :param obj: Object of class :class:`~basyx.aas.model.base.Qualifier`
+        :param parent: The parent node. Can either be :class:`~rdflib.term.URIRef` or :class:`~rdflib.term.BNode`
+        :return: None
+        """
+        qualifier = BNode()
+        self.graph.add((parent, self.aas["Qualifiable/qualifiers"], qualifier))
+        self.graph.add((qualifier, RDF.type, self.aas["Qualifier"]))
+        self._abstract_classes_to_rdf(obj, qualifier)
+        self.graph.add((qualifier, self.aas["Qualifier/kind"], self.aas[f"QualifierKind/{_generic.QUALIFIER_KIND[obj.kind]}"]))
+        self.graph.add((qualifier, self.aas["Qualifier/type"], Literal(str(obj.type), datatype=XSD.string)))
+        self.graph.add((qualifier, self.aas["Qualifier/valueType"], self.aas[f"DataTypeDefXsd/{model.datatypes.XSD_TYPE_NAMES[obj.value_type]}"]))
+        if obj.value:
+            self._value_to_rdf(obj.value, obj.value_type, qualifier, self.aas["Qualifier/value"])
+        if obj.value_id:
+            self._reference_to_rdf(obj.value_id, qualifier, self.aas["Qualifier/valueId"])
+
+    def _extension_to_rdf(self, obj: model.Extension, parent: Union[URIRef, BNode]) -> None:
+        """
+        Serialization of objects of class :class:`~basyx.aas.model.base.Extension` to
+        a joint rdflib Graph object
+
+        :param obj: Object of class :class:`~basyx.aas.model.base.Extension`
+        :param parent: The parent node. Can either be :class:`~rdflib.term.URIRef` or :class:`~rdflib.term.BNode`
+        :return: None
+        """
+        extension = BNode()
+        self.graph.add((parent, self.aas["HasExtensions/extensions"], extension))
+        self.graph.add((extension, RDF.type, self.aas["Extension"]))
+        self._abstract_classes_to_rdf(obj, extension)
+        self.graph.add((extension, self.aas["Extension/name"], Literal(str(obj.name), datatype=XSD.string)))
+        if obj.value_type:
+            self.graph.add((extension, self.aas["Extension/valueType"], self.aas[f"DataTypeDefXsd/{model.datatypes.XSD_TYPE_NAMES[obj.value_type]}"]))
+        if obj.value:
+            self._value_to_rdf(obj.value, obj.value_type, extension, self.aas["Extension/value"])
```

Comment:
I can't figure it out myself, at least not in due time. Because of that, I propose this:

```suggestion
            # Todo: Figure out why mypy complains about this function call and not about others
            self._value_to_rdf(obj.value, obj.value_type, extension, self.aas["Extension/value"])  # type: ignore
```